src/sentry/utils/safe.py

```diff
@@ -22,7 +22,7 @@ def safe_execute(func, *args, **kwargs):
     try:
         if _with_transaction:
             with sentry_sdk.start_span(op="db.safe_execute", description="transaction.atomic"):
-                with transaction.atomic(), django_test_transaction_water_mark():
+                with transaction.atomic("default"), django_test_transaction_water_mark():
```

Comment:
This looks safe...but that's sort of incidental. Anything we can do to clarify/assert this?

Reply:
Broadly, we'll rely on my transaction helpers (which enforce open transactions match the objects they act on) and split db mode to validate this into the future.  There are definitely callsites that may need to change.  Honestly, this helper is really bad and likely needs to get stripped over time.